content/2017.02.handling-python-projects-dependencies.rst

```diff
@@ -102,66 +103,94 @@ Then you can install your program later using this file as a
+i.e. "pinning" them to a specific version.
 
-Because you are using a contrain file you are safe for future updates
-that might break your code.
+You don't need to do so because you are using a contraint file, you
+are safe for future updates that might break your code.
 
-On your CI not using a contrain file will help you to detect that you
-need to take some actions to support the new released version.
+On your CI, don't use a contraint file. It will help you to detect
+that you need to take some actions to support the new released version
+that break your tests.
 
-There are some cases when you still want to pin some version:
+However with that in mind, there are some cases when you still want to
+pin some version:
 
 
 If you know that your project will not work with a lower version
 ----------------------------------------------------------------
 
-If you are using a feature or API that didn't existed before or you
-had a bug that was fixed after.
+If you are using a feature or API that didn't exist before or you hit
+a bug that was fixed later.
 
 .. code-block::
 
    psycopg > 2.5
    colander >= 1.3.2
    ujson >= 1.35
 
+It won't change the way pip handles your dependency, because even if
+you don't put this, pip will always try to install the latest version.
+
+However it will detect if another library or the project using your
+library is trying to use it with a lower version that won't work with
+your code.
+
 
 If you know that your project doesn't support yet the next release
 ------------------------------------------------------------------
 
-**I insist that this implies that the release already exists and that
-your test suite doesn't run on it.**
+**I insist that this applies only if a new version version of a
+dependency has already been released and that your test suite doesn't
+run on it.**
 
-In that case, and only in that case, you can do it for the shorted
-possible time until you ported your project to it.
+In that case, and only in that case, you can pin the dependency's
+version for the shortest possible time until you port your project to
+it.
 
-It is often the case for frameworks:
+It's common to encounter breaking changes when upgrading frameworks:
 
 .. code-block::
 
    Pyramid < 1.8
    django >1.6,<= 1.8
 
-The danger of doing it is that you migh create conflict and in that
-case Python will not let you start your application.
+The danger of doing it is that you might create
+``pkg_resources.VersionConflict`` errors.
+
+When Python starts and imports your lib it will look at the
+requirements list and validate that all dependencies are installed
+with their expected version. If it is not the case Python will not let
+you start your application.
 
-An easy way to break things is to ping a max ``requests`` version for instance:
+However when you install a dependency, pip will check if it is already
+installed without validating if the expected version is installed but
+rather if a version is installed.
```

Comment:
Nit: `any version` might be good, to emphasize that no specific version is checked.